### PR TITLE
ASA Go: Profile tab

### PR DESCRIPTION
**Pull request by conbrad**
This pull request introduces significant updates to the `mobile/asa-go` project, including the addition of new components for elevation-related UI features, updates to existing functionality, and improvements to test coverage. The changes focus on enhancing the `Profile` section, adding visual elements to represent elevation data, and improving maintainability through modularization.

#### New Components for Elevation Features:
* **`CriticalHours` Component**: Added a new component to display critical hours with formatted time ranges. Includes comprehensive unit tests to handle various edge cases and ensure proper styling. [[1]](diffhunk://#diff-9ffa6aa3db3d46e52f4bcbf8db803546b8c5b8e325de4165103fde9bd04c7aa1R1-R25) [[2]](diffhunk://#diff-358ff740a3c937f0894570b384ebfb079ac79ba30cf3568f8d76ecafcfcf8cfeR1-R76)
* **`ElevationFlag` Component**: Added a reusable flag component to visually represent percentages. Includes unit tests to verify integration with mock data and correct rendering of percentages and styles. [[1]](diffhunk://#diff-91f38363ec8af68e22ac41e3f0a73017150fa977ba7589ca1b956a2ebe9be0ebR1-R22) [[2]](diffhunk://#diff-15d4242e7d355a73d687f8c1a29ef0010079661d73e959bb1c2fb8b7393e676aR1-R75)
* **`ElevationLabel` Component**: Added a label component to display elevation zone names with bold styling. Includes tests to handle different label values, special characters, and empty strings. [[1]](diffhunk://#diff-9be81c85bd06349a96f25504168753a6523799f43d583b8f606b5102babe066bR1-R23) [[2]](diffhunk://#diff-f72d63262e7446021bbd8b8ac6a1b1a65dc81098e7ada3362d4b0e282bd5edcfR1-R57)
* **`ElevationStatus` Component**: Introduced a comprehensive component to display elevation data, including labels, flags, and a mountain background. Includes logic to calculate percentages and tests for layout, styling, and edge cases. [[1]](diffhunk://#diff-51b69c1df0c5fcbb19d4d16647b5f3ba96a273d1aa9a9e5cfb12738976d8b8a0R1-R85) [[2]](diffhunk://#diff-4a0258a6e26599a7beb7a74e1bc333e3521cb2029ebf2beae1c7ccedd7160d18R1-R197)

#### Updates to the `Profile` Section:
* **Integration of New Components**: Updated the `Profile` section to include new elevation-related components (`CriticalHours`, `ElevationFlag`, `ElevationLabel`, and `ElevationStatus`) with properties for advisory thresholds, dates, and fire zone data.
* **Removal of Old Profile Component**: Deleted the outdated `Profile` component to replace it with the new modularized implementation.

#### General Enhancements:
* **Dependency Addition**: Added `@mui/x-data-grid-pro` to `package.json` to support advanced Material-UI grid features.
* **Code Cleanup**: Removed unused imports and constants, such as `PST_UTC_OFFSET`, and replaced a dynamic date with a static ISO date for testing purposes. [[1]](diffhunk://#diff-845416c781986e4d3ce7432f5d33c530789e21a164794fc5aae505bf7ac1a6a8L5-R5) [[2]](diffhunk://#diff-845416c781986e4d3ce7432f5d33c530789e21a164794fc5aae505bf7ac1a6a8L21-R29) [[3]](diffhunk://#diff-845416c781986e4d3ce7432f5d33c530789e21a164794fc5aae505bf7ac1a6a8L43-R44)

These changes collectively improve the user interface, modularity, and test coverage of the `Profile` section, making it more maintainable and user-friendly.
# Test Links:
[Landing Page](https://wps-pr-4708-e1e498-dev.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-4708-e1e498-dev.apps.silver.devops.gov.bc.ca/morecast)
[Percentile Calculator](https://wps-pr-4708-e1e498-dev.apps.silver.devops.gov.bc.ca/percentile-calculator)
[C-Haines](https://wps-pr-4708-e1e498-dev.apps.silver.devops.gov.bc.ca/c-haines)
[FireCalc](https://wps-pr-4708-e1e498-dev.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireCalc bookmark](https://wps-pr-4708-e1e498-dev.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN,s=9999&f=c7&c=NaN&w=NaN)
[Auto Spatial Advisory (ASA)](https://wps-pr-4708-e1e498-dev.apps.silver.devops.gov.bc.ca/auto-spatial-advisory)
[HFI Calculator](https://wps-pr-4708-e1e498-dev.apps.silver.devops.gov.bc.ca/hfi-calculator)
[SFMS Insights](https://wps-pr-4708-e1e498-dev.apps.silver.devops.gov.bc.ca/insights)
[Fire Watch](https://wps-pr-4708-e1e498-dev.apps.silver.devops.gov.bc.ca/fire-watch)

### CODE REVIEW

File: mobile/asa-go/src/App.tsx [hunk 1]
```diff
@@ -40,7 +40,8 @@ const App = () => {
     FireShape | undefined
   >(undefined);
   const [dateOfInterest, setDateOfInterest] = useState(
-    DateTime.now().setZone(`UTC${PST_UTC_OFFSET}`)
+    // DateTime.now().setZone(`UTC${PST_UTC_OFFSET}`)
+    DateTime.fromISO("2025-06-15")
```
Comment (brettedw) [inline mobile/asa-go/src/App.tsx:44]: The copilot PR description says this is intentional 😂

Comment (conbrad) [reply mobile/asa-go/src/App.tsx]: Fixed in: https://github.com/bcgov/wps/pull/4708/commits/9e2ba2ffec307690648c8e9b1eb143b49d9748cb

File: mobile/asa-go/src/App.tsx [hunk 2]
```diff
@@ -26,7 +26,7 @@ import { DateTime } from "luxon";
 import { useEffect, useState } from "react";
 import { useDispatch, useSelector } from "react-redux";
 
-const ADVISORY_THRESHOLD = 20;
+const ADVISORY_THRESHOLD = 5;
```
Comment (brettedw) [inline mobile/asa-go/src/App.tsx:29]:
```suggestion
const ADVISORY_THRESHOLD = 20;
```

File: mobile/asa-go/src/components/profile/FuelSummary.tsx [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+  //   }
+  //   const allFilteredZoneUnitFuelStats =
+  //     filteredFireCentreHFIFuelStats[selectedFireCenter.name];
+  //   return (
+  //     allFilteredZoneUnitFuelStats?.[selectedFireZoneUnit.fire_shape_id] ?? {
+  //       fuel_area_stats: [],
+  //       min_wind_stats: [],
+  //     }
+  //   );
+  // }, [filteredFireCentreHFIFuelStats, selectedFireZoneUnit]);
```
[the remark this answers is not in the document]
Comment (conbrad) [reply mobile/asa-go/src/components/profile/FuelSummary.tsx]: I guess so, gone here: https://github.com/bcgov/wps/pull/4708/commits/4f0cd5a616ef1f13408c42277b5476eede83c801

File: mobile/asa-go/src/components/profile/FuelSummary.tsx [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+  DataGridPro,
+  GridColDef,
+  GridColumnHeaderParams,
+  GridRenderCellParams,
+} from "@mui/x-data-grid-pro";
+import { styled } from "@mui/material/styles";
+import CriticalHours from "@/components/profile/CriticalHours";
+import FuelDistribution from "@/components/profile/FuelDistribution";
+// import { useSelector } from 'react-redux'
+// import { selectFilteredFireCentreHFIFuelStats } from '@/slices/fireCentreHFIFuelStatsSlice'
```
[the remark this answers is not in the document]
Comment (conbrad) [reply mobile/asa-go/src/components/profile/FuelSummary.tsx]: https://github.com/bcgov/wps/pull/4708/commits/4f0cd5a616ef1f13408c42277b5476eede83c801

File: mobile/asa-go/src/components/profile/FuelSummary.tsx [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+              sortModel: [{ field: "area", sort: "desc" }],
+            },
+          }}
+          rows={fuelTypeInfoRollup}
+          showCellVerticalBorder
+          showColumnVerticalBorder
+          sx={{
+            backgroundColor: "white",
+            // maxHeight: '147px',
+            // minHeight: '100px',
```
Comment (brettedw) [inline mobile/asa-go/src/components/profile/FuelSummary.tsx:171]: //

Comment (conbrad) [reply mobile/asa-go/src/components/profile/FuelSummary.tsx]: https://github.com/bcgov/wps/pull/4708/commits/1c1aec56ae961a8113c9929d2c21b6f6ab2f6410

File: mobile/asa-go/src/components/profile/FillableFlag.tsx [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+        d="M10.7443 41.822L0.558603 21.375L10.7443 0.928009L119.5 0.928009L119.5 41.822L10.7443 41.822Z"
+        stroke="black"
+        fill="black"
+        mask={`url(#mask-${maskId})`}
+      />
+      <path
+        d="M10.7443 41.822L0.558603 21.375L10.7443 0.928009L119.5 0.928009L119.5 41.822L10.7443 41.822Z"
+        stroke="black"
+      />
+      <text
```
Comment (dgboss) [inline mobile/asa-go/src/components/profile/FillableFlag.tsx:65]: Nice work stuffing the text into here!

File: mobile/asa-go/src/components/profile/FuelSummary.tsx [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+    headerClassName: "fuel-summary-header",
+    headerName: "Fuel Type",
+    sortable: false,
+    minWidth: 120,
+    renderHeader: (params: GridColumnHeaderParams) => (
+      <StyledHeader>{params.colDef.headerName}</StyledHeader>
+    ),
+    renderCell: (params: GridRenderCellParams) => (
+      <Tooltip followCursor placement="right" title={params.row["description"]}>
+        <Typography sx={{ fontSize: "0.75rem", display: "flex", flexGrow: 1 }}>
```
Comment (dgboss) [inline mobile/asa-go/src/components/profile/FuelSummary.tsx:52]: If we add `height: "100%", alignItems: "center"` to the Typography sx prop the text should be centered vertically.

File: mobile/asa-go/src/components/profile/CriticalHours.tsx [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+const CriticalHours = ({ start, end }: CriticalHoursProps) => {
+  let formattedCriticalHours = '-'
+
+  if (!isNil(start) && !isNil(end)) {
+    const [formattedStartTime, formattedEndTime] = formatCriticalHoursTimeText(start, end)
+    formattedCriticalHours = `${formattedStartTime} - ${formattedEndTime}`
+  }
+  return (
+    <Typography sx={{ fontSize: '0.75rem' }} data-testid="critical-hours">
```
Comment (dgboss) [inline mobile/asa-go/src/components/profile/CriticalHours.tsx:19]: If we add `display: "flex", height: "100%", alignItems: "center"` to the Typography sx prop the text should be centered vertically.

File: mobile/asa-go/src/components/profile/FuelSummary.tsx (hunk 7 above)
Comment (conbrad) [reply mobile/asa-go/src/components/profile/FuelSummary.tsx:52]: https://github.com/bcgov/wps/pull/4708/commits/f465787707ea56ae6c04e22b9d6cafc360fa1668

File: mobile/asa-go/src/components/profile/CriticalHours.tsx (hunk 8 above)
Comment (conbrad) [reply mobile/asa-go/src/components/profile/CriticalHours.tsx]: https://github.com/bcgov/wps/pull/4708/commits/f465787707ea56ae6c04e22b9d6cafc360fa1668

File: mobile/asa-go/src/components/profile/FillableFlag.tsx [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+      viewBox="0 0 120 43"
+      fill="none"
+      xmlns="http://www.w3.org/2000/svg"
+      role="img"
+    >
+      <defs>
+        <mask id={`mask-${maskId}`}>
+          <rect x="0" y="0" width={`${fillWidth}`} height="43" fill="white" />
+        </mask>
+        <filter id={`shadow-${maskId}`}>
```
Comment (brettedw) [inline mobile/asa-go/src/components/profile/FillableFlag.tsx:24]: Very minor: Any easy way to keep the text shadow more subtle around the text like the web version?

File: mobile/asa-go/src/components/profile/ElevationStatus.tsx [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+            }}
+          >
+            Portion under advisory:
+          </Typography>
+        </Grid>
+      </Grid>
+      <Grid size={12}>
+        <Box
+          sx={{
+            background: `url(${Mountain})`,
```
Comment (brettedw) [inline mobile/asa-go/src/components/profile/ElevationStatus.tsx:58]: Maybe a later issue, on tablet we get duplicate mountain svg's. 
Unrelated: I think on tablet we could pretty significantly increase the font size in the table as well

Comment (conbrad) [reply mobile/asa-go/src/components/profile/ElevationStatus.tsx:58]: https://github.com/bcgov/wps/issues/4710

File: mobile/asa-go/src/components/profile/FuelSummary.test.tsx [hunk 11]
```diff
@@ -0,0 +1,294 @@
...
+  default: ({ start, end }: { start?: number; end?: number }) => (
+    <div data-testid="critical-hours">
+      {start !== undefined && end !== undefined
+        ? `${start}:00 - ${end}:00`
+        : "-"}
+    </div>
+  ),
+}));
+
+// Mock lodash groupBy function - simplified approach
```
Comment (brettedw) [inline mobile/asa-go/src/components/profile/FuelSummary.test.tsx:47]: Do you think it's necessary to mock `groupBy`?

Comment (conbrad) [reply mobile/asa-go/src/components/profile/FuelSummary.test.tsx]: Nope, just some AI slop, removed: https://github.com/bcgov/wps/pull/4708/commits/1d03570e0e8d354b7d879a1eb4bcfc7f19783433

**Review by dgboss — APPROVED** [review]
Thanks for finishing this one up!

**Review by brettedw — APPROVED** [review]
Nice! I think later it might be worth exploring something other than tooltip's for mobile devices, they aren't really accessible in reasonable way on touch screens
